src/lib/db.ts

```diff
@@ -1,19 +1,39 @@
 import { readFile } from "node:fs/promises";
 import { PromisedDatabase } from "promised-sqlite3";
+import { ASSETDB_DIR } from "../util/config";
+
+let db: PromisedDatabase; 
 
 export async function getConnection(
-  assetDbDir: string = process.env.ASSETDB_DIR as string
+  assetDbDir: string = ASSETDB_DIR
 ): Promise<PromisedDatabase> {
-  const filename = await getDbLocation(assetDbDir);
+  if(!db) {
```

Comment:
I think if you're going to cache the db connection, then you should re-open a new connection if the db location changes (because the asset db was updated, which I would expect to happen via cron).

Reply:
ahh, good catch. Reverting.